benches/big_tree.rs

```diff
@@ -45,39 +40,70 @@ fn build_deep_tree(taffy: &mut Taffy, rng: &mut ChaCha8Rng, max_nodes: u32) -> V
     // Each child gets an equal amount of the remaining nodes
     (0..branching_factor)
         .map(|_| {
-            let sub_children = build_deep_tree(taffy, rng, (max_nodes - branching_factor) / branching_factor);
+            let sub_children =
+                build_deep_tree(taffy, rng, (max_nodes - branching_factor) / branching_factor, branching_factor);
             taffy.new_with_children(FlexboxLayout::random(rng), &sub_children).unwrap()
         })
         .collect()
 }
 
 /// A tree with a higher depth for a more realistic scenario
-fn build_deep_hierarchy(taffy: &mut Taffy) -> Node {
+fn build_deep_hierarchy(taffy: &mut Taffy, node_count: u32, branching_factor: u32) -> Node {
     let mut rng = ChaCha8Rng::seed_from_u64(12345);
 
-    let tree = build_deep_tree(taffy, &mut rng, NODE_COUNT);
+    let tree = build_deep_tree(taffy, &mut rng, node_count, branching_factor);
 
     taffy.new_with_children(FlexboxLayout { ..Default::default() }, &tree).unwrap()
 }
 
 fn taffy_benchmarks(c: &mut Criterion) {
     // Decrease sample size, because the tasks take longer
-    let mut group = c.benchmark_group("big tree (100_000 nodes)");
+    let mut group = c.benchmark_group("big trees");
     group.sample_size(10);
 
-    group.bench_function("flat hierarchy", |b| {
+    group.bench_function("10_000 nodes (2-level hierarchy)", |b| {
+        let mut taffy = Taffy::new();
+        let root = build_flat_hierarchy(&mut taffy, 10_000);
+
+        b.iter(|| taffy.compute_layout(root, Size::NONE).unwrap())
+    });
+
+    group.bench_function("100_000 nodes (2-level hierarchy)", |b| {
+        let mut taffy = Taffy::new();
+        let root = build_flat_hierarchy(&mut taffy, 100_000);
+
+        b.iter(|| taffy.compute_layout(root, Size::NONE).unwrap())
+    });
+
+    group.bench_function("100_000 nodes (7-level hierarchy)", |b| {
         let mut taffy = Taffy::new();
-        let root = build_flat_hierarchy(&mut taffy);
+        let root = build_deep_hierarchy(&mut taffy, 100_000, 7);
 
         b.iter(|| taffy.compute_layout(root, Size::NONE).unwrap())
     });
 
-    group.bench_function("deep hierarchy", |b| {
+    group.bench_function("4000 nodes (12-level hierarchy)", |b| {
         let mut taffy = Taffy::new();
-        let root = build_deep_hierarchy(&mut taffy);
+        let root = build_deep_hierarchy(&mut taffy, 4000, 2);
```

Comment:
Not sure if I fully understand "branching factor", but should this `2` be a `12`?

Reply:
Nope, we're intending to have few children (and thus deep trees).

Reply:
Ah, that's what this does. Gotcha!